visual-tests/collider-mesh/collider-mesh-test.cpp

```diff
@@ -0,0 +1,194 @@
+    });
+    sceneView.Add(model);
+
+    model.ResourceReadySignal().Connect(this, &ColliderMeshTest::LoadingReady );
+    window.Add(sceneView);
+  }
+
+private:
+
+  static bool OnModelHit(Scene3D::Model model, Scene3D::ModelNode node)
+  {
+    static const std::vector<std::string> NAMES = {"room0", "room1", "room2"};
+    for(auto name : NAMES)
+    {
+      auto actor = model.FindChildModelNodeByName(name);
+      actor.SetProperty(Actor::Property::COLOR, Color::WHITE);
+    }
+    node.SetProperty(Actor::Property::COLOR, Color::RED);
+    return true;
+  }
+
+
+  void LoadingReady(Control control)
+  {
+    using Scene3D::Model;
+    auto model = Model::DownCast(control);
+
+    auto camera= model.GenerateCamera(0);
+
+    mSceneView.AddCamera(camera);
+    mSceneView.SelectCamera(mSceneView.GetCameraCount()-1);
+
+    [[maybe_unused]] auto transform = Dali::DevelActor::GetWorldTransform(camera);
+
+    [[maybe_unused]] auto room0 = model.FindChildModelNodeByName("room0");
+    [[maybe_unused]] auto room1 = model.FindChildModelNodeByName("room1");
+    [[maybe_unused]] auto room2 = model.FindChildModelNodeByName("room2");
+
+    auto collider0 = Scene3D::Loader::NavigationMeshFactory::CreateFromFile(TEST_RESOURCES_DIR "collider-mesh/room0.col");
+    auto collider1 = Scene3D::Loader::NavigationMeshFactory::CreateFromFile(TEST_RESOURCES_DIR "collider-mesh/room1.col");
+    auto collider2 = Scene3D::Loader::NavigationMeshFactory::CreateFromFile(TEST_RESOURCES_DIR "collider-mesh/room2.col");
+
+    room0.SetColliderMesh(std::move(collider0));
+    room1.SetColliderMesh(std::move(collider1));
+    room2.SetColliderMesh(std::move(collider2));
+
+    model.MeshHitSignal().Connect(this, &ColliderMeshTest::OnModelHit );
+    PerformNextTest(1000);
+  }
+
+  bool OnUpdate()
+  {
+    if(gTestStep == 1) {
+      Dali::TouchPoint p(0, Dali::PointState::DOWN, 400, 400);
+      EmitTouchEvent(p);
+    }
+    else if(gTestStep == 2)
+    {
+      Dali::TouchPoint p(0, Dali::PointState::DOWN, 798, 511);
+      EmitTouchEvent(p);
+    }
+    else if(gTestStep == 3)
+    {
+      Dali::TouchPoint p(0, Dali::PointState::DOWN, 568, 238);
+      EmitTouchEvent(p);
+    }
+    else if(gTestStep == 4)
+    {
+      mApplication.Quit();
+    }
+    CaptureWindow( mWindow);
+    gTestStep++;
+    return false;
+  }
+
+
+  Dali::Timer mTimer;
+
+  void PerformNextTest(uint32_t millisDelay = 0)
+  {
+    mTimer.Reset();
+    // check previous step
+    mTimer = Dali::Timer::New(1000);
+    mTimer.TickSignal().Connect(this, &ColliderMeshTest::OnUpdate);
+    mTimer.Start();
+  }
+
+  void PostRender() override
+  {
+
+    std::string fname(IMAGE_FILE);
+    fname += std::to_string(gTestStep+1);
+    fname += ".png";
+    CheckImage(fname);
+    PerformNextTest(1000);
+  }
+
+private:
+
+  Application& mApplication;
+  Dali::Window mWindow;
+  Scene3D::SceneView mSceneView;
+};
+
+DALI_VISUAL_TEST_WITH_WINDOW_SIZE( ColliderMeshTest, OnInit, 1280, 720 )
```

Comment:
Are all the images and textures copyright free?

Reply:
I guess so. I generated them myself.